File: remote/src/test/java/com/kelsos/mbrc/mappers/QueueTrackMapperTest.java
package com.kelsos.mbrc.mappers;

import com.kelsos.mbrc.domain.QueueTrack;
import com.kelsos.mbrc.dto.NowPlayingTrack;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

import static junit.framework.TestCase.assertEquals;

public class QueueTrackMapperTest {
  public static final String TRACK_TITLE_1 = "Dialogue With the Stars";
  public static final String TRACK_ARIST_1 = "In Flames";
  public static final String TRACK_PATH_1 = "K:\\media\\Melodic Death Metal\\In Flames\\1997 - Whoracle\\09 - Episode 666.mp3";
  private static final String TRACK_TITLE_2 = "At the Gates of Sleep";
  public static final String TRACK_ARIST_2 = "Insomnium";
  public static final String TRACK_PATH_2 = "K:\\media\\Melodic Death Metal\\Insomnium\\2006-10-17 - Above the Weeping World\\05 - At the Gates of Sleep.mp3";


  private List<NowPlayingTrack> tracks;

  @Before
  public void setUp() throws Exception {
    tracks = new ArrayList<>();
    final NowPlayingTrack track = new NowPlayingTrack();
    track.setTitle(TRACK_TITLE_1);
    track.setArtist(TRACK_ARIST_1);
    track.setPath(TRACK_PATH_1);
    track.setPosition(1);
    tracks.add(track);

    final NowPlayingTrack track2 = new NowPlayingTrack();
    track2.setTitle(TRACK_TITLE_2);
    track2.setArtist(TRACK_ARIST_2);
    track2.setPath(TRACK_PATH_2);
    track2.setPosition(2);
    tracks.add(track2);
  }

  @After
  public void tearDown() throws Exception {
    tracks.clear();
  }

  @Test
  public void testMapSingle() throws Exception {
    final QueueTrack track = QueueTrackMapper.map(tracks.get(0));

    assertEquals(TRACK_TITLE_1, track.getTitle());
    assertEquals(TRACK_ARIST_1, track.getArtist());
    assertEquals(TRACK_PATH_1, track.getPath());
    assertEquals(1, track.getPosition());
  }

  @Test
  public void testMapMultiple() throws Exception {
    final List<QueueTrack> tracks = QueueTrackMapper.map(this.tracks);

    final QueueTrack track = tracks.get(1);
    assertEquals(2, tracks.size());
    assertEquals(TRACK_ARIST_2, track.getArtist());
    assertEquals(TRACK_TITLE_2, track.getTitle());
    assertEquals(TRACK_PATH_2, track.getPath());
    assertEquals(2, track.getPosition());
  }
}

File: app/src/test/kotlin/com/kelsos/mbrc/mappers/QueueTrackMapperTest.kt
package com.kelsos.mbrc.mappers

import com.kelsos.mbrc.dto.NowPlayingTrack
import org.junit.After
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import java.util.*

class QueueTrackMapperTest {

    private var tracks: MutableList<NowPlayingTrack>? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        tracks = ArrayList<NowPlayingTrack>()
        val track = NowPlayingTrack()
        track.title = TRACK_TITLE_1
        track.artist = TRACK_ARIST_1
        track.path = TRACK_PATH_1
        track.position = 1
        tracks!!.add(track)

        val track2 = NowPlayingTrack()
        track2.title = TRACK_TITLE_2
        track2.artist = TRACK_ARIST_2
        track2.path = TRACK_PATH_2
        track2.position = 2
        tracks!!.add(track2)
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
        tracks!!.clear()
    }

    @Test
    @Throws(Exception::class)
    fun testMapSingle() {
        val track = QueueTrackMapper.map(tracks!![0])

        Assert.assertEquals(TRACK_TITLE_1, track.title)
        Assert.assertEquals(TRACK_ARIST_1, track.artist)
        Assert.assertEquals(TRACK_PATH_1, track.path)
        Assert.assertEquals(1, track.position)
    }

    @Test
    @Throws(Exception::class)
    fun testMapMultiple() {
        val tracks = QueueTrackMapper.map(this.tracks!!)

        val track = tracks[1]
        Assert.assertEquals(2, tracks.size)
        Assert.assertEquals(TRACK_ARIST_2, track.artist)
        Assert.assertEquals(TRACK_TITLE_2, track.title)
        Assert.assertEquals(TRACK_PATH_2, track.path)
        Assert.assertEquals(2, track.position)
    }

    companion object {
        val TRACK_TITLE_1 = "Dialogue With the Stars"
        val TRACK_ARIST_1 = "In Flames"
        val TRACK_PATH_1 = "K:\\media\\Melodic Death Metal\\In Flames\\1997 - Whoracle\\09 - Episode 666.mp3"
        private val TRACK_TITLE_2 = "At the Gates of Sleep"
        val TRACK_ARIST_2 = "Insomnium"
        val TRACK_PATH_2 = "K:\\media\\Melodic Death Metal\\Insomnium\\2006-10-17 - Above the Weeping World\\05 - At the Gates of Sleep.mp3"
    }
}
